programs/cleos/main.cpp

```diff
@@ -1010,14 +1018,23 @@ struct create_account_subcommand {
       createAccount->set_callback([this] {
             if( !active_key_str.size() )
                active_key_str = owner_key_str;
-            public_key_type owner_key, active_key;
+            auth_type owner, active;
             try {
-               owner_key = public_key_type(owner_key_str);
-            } EOS_RETHROW_EXCEPTIONS(public_key_type_exception, "Invalid owner public key: ${public_key}", ("public_key", owner_key_str));
+               owner = public_key_type(owner_key_str);
+            } catch (...) {
+               try {
+                  owner = get_permission_level(owner_key_str);
+               } EOS_RETHROW_EXCEPTIONS(public_key_type_exception, "Invalid owner public key: ${public_key}", ("public_key", owner_key_str));
```

Comment:
I think we need to clean this error message up to include permissions, also it seems like `get_permission_level` will throw `name_type_exception` if the names are bad or generically assert if there is no separator.  Both of which will not get caught as a `public_key_type_exception`